File: src/main/scala/labs/lab2/TalkingRobot.scala
package labs.lab2

/**
  * @author Evgeny Borisov
  */
object TalkingRobot {
  var quoters:List[Quoter] =  List(RandomQuoter,
    new ShakespearQuoter(),
    MessageQuoter("Good to die for our country"))

  def addQuoter(quoter: Quoter):Unit={
    quoters = quoter::quoters
  }

  def talk():Unit={
    quoters.foreach(_.printMessage())
  }
}

File: src/main/java/real_spring/TalkingRobot.java
package real_spring;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import java.util.List;

/**
 * @author Evgeny Borisov
 */
@Component
public class TalkingRobot {
    @Autowired
    private List<Quoter> quoters;


//    @Scheduled(cron = "1/1 * * * * ? ")
    @Scheduled(fixedDelay = 1000)
    public void talk() {
        quoters.forEach(Quoter::sayQuote);
    }
}
